src/server.rs

```diff
@@ -39,7 +40,7 @@ impl Server {
 
 pub fn build_http_request(mut stream: TcpStream) -> (TcpStream, Request) {
     let (request, body) = parse_tcp_stream(&mut stream);
-    let request = parse_http_string((request, body));
+    let request = Request::from_str(&request).unwrap().body(body).clone();
```

Comment:
Implementing from_str is a good idea.
However 
```rust
(&request).unwrap().body(body).clone();
```
is excatly why I don't like chainable methods.
I feel like it's hurting readability here.

Why would it be more preferable to call several methods on a struct instead of just initializing it normaly?

Aslo why are you cloning? Not really seeing a use case for it here.

Reply:
Chainable methods is needed here, You can't passing any arguments other than string that you want parse, So It's important here and unavoidable (Unless there is a way that I don't know) if we want use [from_str](https://doc.rust-lang.org/std/str/trait.FromStr.html#tymethod.from_str)